xapian-core/queryparser/termgenerator_internal.cc

```diff
@@ -95,6 +95,53 @@ check_infix(unsigned ch)
     return 0;
 }
 
+static inline bool
+is_ascii_digit(unsigned ch)
+{
+    return (ch < 128 && C_isdigit(static_cast<unsigned char>(ch)));
```

Comment:
While the outer `(` ... `)` are OK here, they aren't required by the language and result in a stacked `)))` at the end, which makes mentally matching parentheses a bit harder.

As a rule of thumb, parentheses are probably better only used when they are required by the language, when you want to control the relative order of evaluation (e.g. `a * b + c` vs `a * (b + c)` or even `a + b + c` vs `a + (b + c)`), or where they are likely to improve clarity for someone reading the code - e.g. it's probably clearer to write `a || (b && c)` (and compilers tend to warn for `a || b && c` because a lot of people have to think about the precedence order).

Reply:
Thanks, I'll fix it.